pkg/kubelet/kubelet.go

```diff
@@ -1343,6 +1343,14 @@ type Kubelet struct {
 
 	// Track node startup latencies
 	nodeStartupLatencyTracker util.NodeStartupLatencyTracker
+
+	// whether initial static pods registration was completed or not
+	// before node became ready.
+	initialStaticPodsRegistered bool
+
+	// Lock to ensure it is concurrency-safe to access
+	// initialStaticPodsRegistered field.
+	initialStaticPodsRegisteredLock sync.Mutex
```

Comment:
This locks `initialStaticPodsRegistration()` entire execution, that IIUIC is not necessarily to lock the entire operation and we just may be worried about the access to the `initialStaticPodsRegistered` boolean, if that is the case, we can just use `atomic.Bool` and avoid adding more locking